### ISSUE TITLE
PutBucketLifecycleConfiguration invalid ISO date format

### ISSUE THREAD

**Issue by bennettbuchanan** [aws/aws-sdk-js#2352]
## Context
* aws-sdk v2.28.0
* aws-cli v1.10.38

## Problem
Setting a lifecycle configuration with an ISO format expiration date `2016-01-01T00:00:00.000Z` returns `MalformedXML` error using the SDK. As far as I can tell, this should be a valid configuration (see example https://docs.aws.amazon.com/cli/latest/reference/s3api/put-bucket-lifecycle.html#examples). Here is the script I've used:

```js
const AWS = require('aws-sdk');

const s3 = new AWS.S3({
    endpoint: 'https://s3.amazonaws.com',
    accessKeyId: '...',
    secretAccessKey: '...',
    apiVersions: { s3: '2006-03-01' },
    signatureCache: false,
    signatureVersion: 'v4',
    region: 'us-east-1',
    s3ForcePathStyle: true,
    maxRetries: 0,
    httpOptions: { timeout: 0 },
});

const params = {
    Bucket: 'my-bucket',
    LifecycleConfiguration: {
        Rules: [{
            Expiration: { Date: '2016-01-01T00:00:00.000Z' },
            ID: 'Delete 2014 logs in 2016.',
            Prefix: 'logs/2014/',
            Status: 'Enabled',
        }],
    },
}

s3.putBucketLifecycleConfiguration(params, console.log);
```

Response:

```
{ MalformedXML: The XML you provided was not well-formed or did not validate against our published schema
    at Request.extractError (/Users/bennettbuchanan/sandbox/node_modules/aws-sdk/lib/services/s3.js:539:35)
    at Request.callListeners (/Users/bennettbuchanan/sandbox/node_modules/aws-sdk/lib/sequential_executor.js:105:20)
    at Request.emit (/Users/bennettbuchanan/sandbox/node_modules/aws-sdk/lib/sequential_executor.js:77:10)
    at Request.emit (/Users/bennettbuchanan/sandbox/node_modules/aws-sdk/lib/request.js:673:14)
    at Request.transition (/Users/bennettbuchanan/sandbox/node_modules/aws-sdk/lib/request.js:22:10)
    at AcceptorStateMachine.runTo (/Users/bennettbuchanan/sandbox/node_modules/aws-sdk/lib/state_machine.js:14:12)
    at /Users/bennettbuchanan/sandbox/node_modules/aws-sdk/lib/state_machine.js:26:10
    at Request.<anonymous> (/Users/bennettbuchanan/sandbox/node_modules/aws-sdk/lib/request.js:38:9)
    at Request.<anonymous> (/Users/bennettbuchanan/sandbox/node_modules/aws-sdk/lib/request.js:675:12)
    at Request.callListeners (/Users/bennettbuchanan/sandbox/node_modules/aws-sdk/lib/sequential_executor.js:115:18)
  message: 'The XML you provided was not well-formed or did not validate against our published schema',
  code: 'MalformedXML',
  region: null,
  time: 2018-11-15T00:39:03.304Z,
  requestId: '8062B9E8B428232A',
  extendedRequestId: 'fToki6iupz84WOeZYcIk8FKzc9VgnR/WyDL5vgoXeCOGYxtWyZzNIkKHKoIqUYFNCfb58Y9l/Lk=',
  cfId: undefined,
  statusCode: 400,
  retryable: false }
```

Using the same configuration with the AWS-CLI is successful:

```sh
$ cat lifecycle-config.json
{
    "Rules": [
        {
            "Expiration": { "Date": "2016-01-01T00:00:00.000Z" },
            "ID": "Delete 2014 logs in 2016.",
            "Prefix": "logs/2014/",
            "Status": "Enabled"
        }
    ]
}
$ aws s3api put-bucket-lifecycle-configuration --bucket my-bucket --lifecycle-configuration file://lifecycle-config.json
```

## Solution 

It appears the XML request sent to AWS has changed the `Date` field such that it is no longer in ISO 8601 format.

The XML Date field in the PUT request to AWS from the SDK:

```
...<Date>Fri, 01 Jan 2016 00:00:00 GMT</Date>...
```

The XML Date field in the PUT request to AWS from the CLI:

```
...<Date>2016-01-01T00:00:00.000Z</Date>...
```

Thank you for any insight you can offer into this problem!

**Comment by srchase** [comment]
@bennettbuchanan 

This issue was resolved in [2.60.0](https://github.com/aws/aws-sdk-js/blob/HEAD/CHANGELOG.md#2600).  Upgrading to any version 2.60.0 or later will resolve this issue.

**Comment by bennettbuchanan** [comment]
Great, thanks for the quick reply @srchase.
